src/Service/UpdateVideoItem/UpdateVideoItemRequest.php

```diff
@@ -51,9 +51,12 @@ public function __construct(int $mpxId, string $mediaTypeId) {
    *   The request object, to be passed to the UpdateVideoItem service.
    */
   public static function createFromMediaEntity(Media $entity): UpdateVideoItemRequest {
-    $mpx_id = (int) $entity->get('field_mpx_id')->value;
-    $media_type_id = $entity->bundle();
-    return new self($mpx_id, $media_type_id);
+    $media_source = $entity->bundle->entity->getSource();
+    $source_field = $media_source->getSourceFieldDefinition($entity->bundle->entity);
+
+    $global_url_parts = explode('/', $entity->get($source_field->getName())->value);
+    $mpx_id = (int) end($global_url_parts);
```

Comment:
Now that I see this, we are trying to be consistent and use the full IDs. The API makes no guarantees that just the numeric part of the ID is unique, though I think in practice it is. I'd say this class should construct either with a string or with a Uri object like `\Lullabot\Mpx\DataService\DataObjectFactory::load`.

Reply:
@deviantintegral Originally I started passing the Uri object, but then you pointed out that wasn't needed as there's a method to do it with the numeric one in the DataObjectImporter (which is handy, just a shame we can't rely on that from mpx :( ). Since other parts are already relying on this, and in any case this PR does some improvements and a bugfix, can we open an issue for that? That way it stays documented, even if not tackled immediately.

Reply:
Yeah, in fact I have a few followups to file from the demo. Thanks!